cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -69,13 +74,22 @@ void MetricsCollector::Initialize(std::string metric_namespace,
 
 void MetricsCollector::SubscribeAllTopics()
 {
-  for (auto it = topics_.begin(); it != topics_.end(); ++it) {
-    auto sub = node_->create_subscription<ros_monitoring_msgs::msg::MetricList>(
-            *it, kNodeSubQueueSize,
-            [this](ros_monitoring_msgs::msg::MetricList::UniquePtr metric_list_msg) -> void {
+  for (auto it = topics_.cbegin(); it != topics_.cend(); ++it) {
+    std::shared_ptr<rclcpp::SubscriptionBase> sub;
+    if (it->topic_type == TopicType::ROS_MONITORING_MSGS) {
+      sub = node_->create_subscription<ros_monitoring_msgs::msg::MetricList>(
+              it->topic_name, kNodeSubQueueSize,
+              [this](ros_monitoring_msgs::msg::MetricList::UniquePtr metric_list_msg) -> void {
                 this->RecordMetrics(std::move(metric_list_msg));
-            });
-    subscriptions_.push_back(sub);
+              });
+    } else if (it->topic_type == TopicType::METRICS_STATISTICS_MSGS) {
+      sub = node_->create_subscription<metrics_statistics_msgs::msg::MetricsMessage>(
+              it->topic_name, kNodeSubQueueSize,
+              [this](metrics_statistics_msgs::msg::MetricsMessage::UniquePtr msg) -> void {
+                this->RecordMetrics(std::move(msg));
+              });
+    }
+    subscriptions_.push_back(std::move(sub));
```

Comment:
The move is not required here for a shared_ptr.

Reply:
I think it is okay to do so here: https://stackoverflow.com/a/41874953.